app/src/test/java/com/poti/android/presentation/user/editprofile/EditProfileViewModelTest.kt

```
@@ -0,0 +1,202 @@
+            private set
+        var lastPatchedProfileImageUrl: String? = null
+            private set
+
+        fun prepareNicknameResult(nickname: String): CompletableDeferred<Result<Boolean>> =
+            CompletableDeferred<Result<Boolean>>().also { nicknameResults[nickname] = it }
+
+        override suspend fun postNicknameDuplicate(nickname: String): Result<Boolean> {
+            val result = nicknameResults.getValue(nickname)
+            return try {
+                result.await()
+            } catch (cancellation: CancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)